shapeshift/web
# fix(defi): cosmos sdk empty opportunity routing

## Description

Fixes the DeFi routing for Cosmos SDK opportunities in case you're not staked into said validator - we should use the metadata instead of the userdata in this case.

## Notice




## Pull Request Type


## Issue (if applicable)

N/A

## Risk



None

## Testing



- Connect a wallet without any delegations / undelegations / redelegations for the default ShapeShift validator for ATOM or OSMO
- Click the validator in the DeFi earn section
- Ensure the routing is working

### Engineering



- ☝🏽 

### Operations



- ☝🏽 

## Screenshots (if applicable)

<img width="1236" alt="image" src="https://user-images.githubusercontent.com/17035424/219120800-b940db32-14a7-487d-bc37-757e08878ce0.png">
